allennlp/common/file_utils.py

```diff
@@ -617,64 +615,91 @@ def __init__(
 
-            # This is always how lmdb names the lock file.
-            lock_filename = filename + "-lock"
-            if os.path.isfile(lock_filename):
-                use_lock = os.access(lock_filename, os.W_OK)
-            else:
-                # If the lock file doesn't exist yet, then the directory needs to be writable in
-                # order to create and use the lock file.
-                use_lock = os.access(os.path.dirname(lock_filename), os.W_OK)
-
-        if not use_lock:
-            warnings.warn(
-                f"Lacking permissions to use lock file on cache '{filename}'.\nUse at your own risk!",
-                UserWarning,
-            )
+                # This is always how lmdb names the lock file.
+                lock_filename = filename + "-lock"
+                if os.path.isfile(lock_filename):
+                    use_lock = os.access(lock_filename, os.W_OK)
+                else:
+                    # If the lock file doesn't exist yet, then the directory needs to be writable in
+                    # order to create and use the lock file.
+                    use_lock = os.access(os.path.dirname(lock_filename), os.W_OK)
 
-        self.lmdb_env = lmdb.open(
-            filename,
-            subdir=False,
-            map_size=map_size,
-            max_readers=cpu_count * 4,
-            max_spare_txns=cpu_count * 4,
-            metasync=False,
-            sync=True,
-            readahead=False,
-            meminit=False,
-            readonly=read_only,
-            lock=use_lock,
-        )
-        _active_tensor_caches[self.lmdb_env.path()] = self
+            if not use_lock:
+                warnings.warn(
+                    f"Lacking permissions to use lock file on cache '{filename}'.\nUse at your own risk!",
+                    UserWarning,
+                )
 
-        # We have another cache here that makes sure we return the same object for the same key. Without it,
-        # you would get a different tensor, using different memory, every time you call __getitem__(), even
-        # if you call it with the same key.
-        # The downside is that we can't keep self.cache_cache up to date when multiple processes modify the
-        # cache at the same time. We can guarantee though that it is up to date as long as processes either
-        # write new values, or read existing ones.
-        self.cache_cache: MutableMapping[str, Tensor] = WeakValueDictionary()
+            self.lmdb_env = lmdb.open(
+                filename,
+                subdir=False,
+                map_size=map_size,
+                max_readers=cpu_count * 4,
+                max_spare_txns=cpu_count * 4,
+                metasync=False,
+                sync=True,
+                readahead=False,
+                meminit=False,
+                readonly=read_only,
+                lock=use_lock,
+            )
+            _active_tensor_caches[self.lmdb_env.path()] = self
```

Comment:
Where do we remove entries from `_active_tensor_caches`? I see it in an older commit, but not in the latest one.

Reply:
`_active_tensor_caches` is a `WeakValueDict`, which removes entries automatically when the values are GCd.